tests/conftest.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+import json
+import apypie
+import requests
+
+from urllib.parse import urlparse
+from collections import namedtuple
+
+User = namedtuple('User', ['username', 'password', 'name'])
+
+
+def pytest_generate_tests(metafunc):
+    with open(metafunc.config.option.variables[0]) as f:
+        variables = json.load(f)
+    assert variables
+
+    if 'url' in metafunc.fixturenames:
+        base_url = metafunc.config.option.base_url
+        assert base_url
+
+        # mimic the variables fixture from pytest-variables
+        user_obj = _user(metafunc.config._variables)  # pylint: disable=protected-access
+
+        response = requests.get(f'{base_url}/menu', auth=(user_obj.username, user_obj.password),
+                                verify=False)
+        if response.status_code == 404:
+            # Menu is only available since Foreman 2.0
+            pages = [base_url + page for page in PAGES]
+        else:
+            assert response
+            pages = [pytest.param(base_url + page['url'], id=page['name'])
+                     for page in response.json()
+                     # logout has an error "Cannot read property 'icon' of null"
+                     if page['url'] != '/logout']
+
+        metafunc.parametrize('url', pages)
+
+    if 'katello_client' in metafunc.fixturenames:
+        clients = variables.get('clients', [])
+
+        metafunc.parametrize('katello_client', clients, indirect=True)
+
+
+def _user(variables):
+    return User(variables.get('username', 'admin'), variables.get('password', 'changeme'),
+                variables.get('name', 'Admin User'))
+
+
+@pytest.fixture(scope='session')
+def user(variables):
+    return _user(variables)
+
+
+@pytest.fixture(scope='session')
+def entities():
+    return {
+        'organization_label': 'Test_Organization',
+        'activation_key': 'Test AK',
+        'product': 'Test Product',
+        'product_label': 'Test_Product',
+        'yum_repository': 'Zoo',
+        'yum_repository_label': 'Zoo',
+        'errata_id': 'RHEA-2012:0055',
+        'container_repository_label': 'foremanbusybox'
+    }
+
+@pytest.fixture(scope='session')
+def api(user, base_url):
+    api = apypie.Api(
+        uri=base_url,
+        username=user.username,
+        password=user.password,
+        api_version=2,
```

Comment:
Isn't this the default? Shouldn't we rely on that?

Reply:
No, it's default for Foreman, but not for apipie (it has no way to learn what the default is).

Reply:
What happens if the default changes in the future and then this breaks? I mostly borrowed this from FAM, so @evgeni may have an additional reason.